Documentation/kube-aws-approver.md

```diff
@@ -0,0 +1,100 @@
+# Kubernetes AWS approver
+
+A custom node CertificateSigningRequest approver for kubernetes clusters running on AWS.
+
+## Why do I want this?
+
+[kubelet TLS bootstrapping][tls-bootstrapping] allows kubelet to create CertificateSigningRequest for its client certs. These CertificateSigningRequest need to be approved so as to be signed by kubernetes internal certificate signer controller.
+
+Currently you can either manually approve each node that tries to join the cluster. Or, instruct kubernetes internal certificate approver to approve all certificates from `system:bootstrappers` group. The human intervention required for manual approval is not ideal in auto scaling environments. And the default auto approver performs very basic checks that might not be very secure.
+
+`kube-aws-approver` goes a step further from default auto approver and uses the CertificateSigningRequest generated by node to extract information about the node requesting the certificate, and tries to validate it against AWS API before approving the CertificateSigningRequest.
+
+## How does it work?
+
+kube-aws-approver matches incomming CertificateSigningRequest against two critera:
+
+### Criteria 1: Node already belonging to the cluster requests for client certificate renewal.
+
+- The CertificateRequest `Organization` is equal to `system:nodes`
+- The CertificateRequest `DNSNames` list is empty
+- The CertificateRequest `EmailAddresses` list is empty
+- The CertificateRequest `IPAddresses` list is empty
+- The CertificateRequest `CommonName` has `system:node:` prefix
+- The CertificateSigningRequest `Username` matches the CertificateRequest `CommonName`, i.e. the node provided its client certificate to create this CertificateSigningRequest.
+- The CertificateSigningRequest `Groups` has `system:nodes`
+- There exists an `instance` in `running state` corresponding to the `nodename` from the CertificateRequest `CommonName`
+- A `node` with the `nodename` from CertificateRequest CommonName is `part` of the cluster and in `Ready` state.
+- The `instance` should belong to an AutoScalingGroup.
+- The AutoSaclingGroup belongs to the list of whitelisted AutoScalingGroups provided to `kube-aws-approver` at startup.
+
+Only when a CertificateSigningRequest from a node looking to renew its client certificate meets `all` the above requirements, the CertificateSigningRequest is `Approved`
```

Comment:
Does this mean the crt.CertificateSigningRequestStatus is set to `Approved`? Please clarify

Reply:
I don't think being specific as to how a `CertificateSigningRequest` is approved helps here in the README.

Reply:
It helps the user get an idea of what exactly the approver does to "approve" the CSR. Also when the we add the signer component its easier to tie it in.

Reply:
:+1: added a small section